runtime/JavaScript/src/antlr4/Lexer.d.ts

```diff
@@ -16,13 +16,16 @@ export declare class Lexer extends Recognizer<number> {
     _tokenStartLine: number;
     _tokenStartColumn: number;
     _type: number;
+    _modeStack: number[];
+    _mode: number;
```

Comment:
maybe rather add `getMode` an `getModeStack` ?

Reply:
The public fields would be better because I have also used them in my various portings:
[Java port](https://github.com/antlr/grammars-v4/blob/bf950dfcfb26477735f02402dadf98e1e507c051/python/python3_12_1/Java/PythonLexerBase.java#L293)
[JavaScript port](https://github.com/antlr/grammars-v4/blob/bf950dfcfb26477735f02402dadf98e1e507c051/python/python3_12_1/JavaScript/PythonLexerBase.js#L292)
[Pyton port](https://github.com/antlr/grammars-v4/blob/bf950dfcfb26477735f02402dadf98e1e507c051/python/python3_12_1/Python3/PythonLexerBase.py#L238)

Reply:
Well these fields are clearly private... (I guess there was a bit of laziness with the first implementation 30 years ago)